syncapi/notifier/notifier.go

```diff
@@ -282,8 +282,8 @@ func (n *Notifier) IsSharedUser(userA, userB string) bool {
 	defer n.lock.RUnlock()
 	var okA, okB bool
 	for _, users := range n.roomIDToJoinedUsers {
-		_, okA = users[userA]
-		_, okB = users[userB]
+		okA = users.isIn(userA)
+		okB = users.isIn(userB)
```

Comment:
Just a minor thing that I already had seen yesterday, no need to access the map twice if `userA` isn't already in there.
```suggestion
		okA = users.isIn(userA)
		if !okA {
		    continue
		}
		okB = users.isIn(userB)
```